### PR TITLE
Handle unavailable share providers more graceful

### PR DESCRIPTION
**Pull request by rullzer**
If the sharebymail app (or the federatedshareprovider app) are disabled
this kills a lot of stuff. So we should handle a share provider not
begin available a bit more graceful.

To test:

1. disable sharebymail
2. login and try to open files

Before: :boom: 
Now: No :boom: 

@ChristophWurst as discussed

CC: @schiessle @nickvergessen @MorrisJobke @LukasReschke

### CODE REVIEW

**Review by LukasReschke — APPROVED** [review]

**Comment by LukasReschke** [comment]
LGTM

**Comment by MorrisJobke** [comment]
Tested and works 👍

File: lib/private/Share20/Manager.php [hunk 1]
```diff
@@ -1038,7 +1051,10 @@ public function getSharesByPath(\OCP\Files\Node $path, $page=0, $perPage=50) {
...
 	 */
 	public function getShareByToken($token) {
-		$provider = $this->factory->getProviderForType(\OCP\Share::SHARE_TYPE_LINK);
+		try {
+			$provider = $this->factory->getProviderForType(\OCP\Share::SHARE_TYPE_LINK);
+		} catch (ProviderException $e) {
+		}
 
 		try {
 			$share = $provider->getShareByToken($token);
```
Comment (nickvergessen) [inline lib/private/Share20/Manager.php]: provider may be undefined

File: lib/private/Share20/Manager.php [hunk 2]
```diff
@@ -1048,7 +1064,10 @@ public function getShareByToken($token) {
 
 		// If it is not a link share try to fetch a federated share by token
 		if ($share === null) {
-			$provider = $this->factory->getProviderForType(\OCP\Share::SHARE_TYPE_REMOTE);
+			try {
+				$provider = $this->factory->getProviderForType(\OCP\Share::SHARE_TYPE_REMOTE);
+			} catch (ProviderException $e) {
+			}
 			try {
 				$share = $provider->getShareByToken($token);
```
Comment (nickvergessen) [inline lib/private/Share20/Manager.php]: provider may be undefined

File: lib/private/Share20/Manager.php [hunk 3]
```diff
@@ -1058,7 +1077,10 @@ public function getShareByToken($token) {
 
 		// If it is not a link share try to fetch a federated share by token
 		if ($share === null && $this->shareProviderExists(\OCP\Share::SHARE_TYPE_EMAIL)) {
-			$provider = $this->factory->getProviderForType(\OCP\Share::SHARE_TYPE_EMAIL);
+			try {
+				$provider = $this->factory->getProviderForType(\OCP\Share::SHARE_TYPE_EMAIL);
+			} catch (ProviderException $e) {
+			}
 			$share = $provider->getShareByToken($token);
```
Comment (nickvergessen) [inline lib/private/Share20/Manager.php]: provider may be undefined

**Comment by nickvergessen** [comment]
@rullzer @MorrisJobke please fix ^

**Comment by MorrisJobke** [comment]
@rullzer: ping ;)

**Pull request merged**